### PR TITLE
Add new properties to flexibilize requests

### PR DESCRIPTION
**Pull request by jmarchal-nuxeo**

### CODE REVIEW

**Comment by kevinleturc** [comment]
T&P OK: https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-kleturc-10.10/16/

File: nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java [hunk 1]
```diff
@@ -89,7 +96,8 @@
 
     public OpenIDConnectProvider(OAuth2ServiceProvider oauth2Provider, String accessTokenKey, String userInfoURL,
             Class<? extends OpenIDUserInfo> openIdUserInfoClass, String icon, boolean enabled,
-            RedirectUriResolver redirectUriResolver, Class<? extends UserResolver> userResolverClass, String userMapper) {
+            RedirectUriResolver redirectUriResolver, Class<? extends UserResolver> userResolverClass,
+            String userMapper) {
```
Comment (efge) [inline nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java]: Please undo formatting, this should be a separate commit

File: nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java [hunk 2]
```diff
@@ -136,9 +144,9 @@ public String createStateToken(HttpServletRequest request) {
      * @param HttpServletRequest request
      */
     public boolean verifyStateToken(HttpServletRequest request) {
-        return request.getParameter(OpenIDConnectAuthenticator.STATE_URL_PARAM_NAME)
-                      .equals(request.getSession().getAttribute(
-                              OpenIDConnectAuthenticator.STATE_SESSION_ATTRIBUTE + "_" + getName()));
+        return request.getParameter(OpenIDConnectAuthenticator.STATE_URL_PARAM_NAME).equals(
+                request.getSession()
+                       .getAttribute(OpenIDConnectAuthenticator.STATE_SESSION_ATTRIBUTE + "_" + getName()));
```
Comment (efge) [inline nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java]: Undo formatting

File: nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java [hunk 3]
```diff
@@ -207,14 +219,19 @@ public String getAccessToken(HttpServletRequest req, String code) {
     public OpenIDUserInfo getUserInfo(String accessToken) {
         OpenIDUserInfo userInfo = null;
 
-        HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory(request -> request.setParser(new JsonObjectParser(
-                JSON_FACTORY)));
+        HttpRequestFactory requestFactory = HTTP_TRANSPORT.createRequestFactory(
+                request -> request.setParser(new JsonObjectParser(JSON_FACTORY)));
```
Comment (efge) [inline nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java]: Undo formatting

File: nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java (hunk 1 above)
Comment (jmarchal-nuxeo) [reply nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java]: Done

File: nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java (hunk 2 above)
Comment (jmarchal-nuxeo) [reply nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java]: Done

File: nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java (hunk 3 above)
Comment (jmarchal-nuxeo) [reply nuxeo-services/login/nuxeo-platform-login-openid/src/main/java/org/nuxeo/ecm/platform/oauth2/openid/OpenIDConnectProvider.java]: Done
